REGRESSION(272743@main): WebVTT default styling is broken

anttijk opened the pull request:
#### 472c2a9abc39ccee3c44f9bbe1c2f97cc257780d
<pre>
REGRESSION(272743@main): WebVTT default styling is broken
<a href="https://bugs.webkit.org/show_bug.cgi?id=270783">https://bugs.webkit.org/show_bug.cgi?id=270783</a>
<a href="https://rdar.apple.com/124380882">rdar://124380882</a>

Reviewed by Anne van Kesteren.

272743@main changed the attribute name for pseudo elements.
The WebVTT styling relied on those names.

* LayoutTests/media/track/captions-webvtt/inline.vtt: Added.
* LayoutTests/media/track/webvtt-inline-expected.txt: Added.
* LayoutTests/media/track/webvtt-inline.html: Added.

Add a test. We had no coverage.

* Source/WebCore/Modules/modern-media-controls/controls/text-tracks.css:
([useragentpart=&quot;-webkit-media-text-track-display&quot;]):
(u):
(i):
(.hidden):
([pseudo=&quot;-webkit-media-text-track-display&quot;] b): Deleted.
([pseudo=&quot;-webkit-media-text-track-display&quot;] u): Deleted.
([pseudo=&quot;-webkit-media-text-track-display&quot;] i): Deleted.
([pseudo=&quot;-webkit-media-text-track-display&quot;] .hidden): Deleted.
* Source/WebCore/css/mediaControls.css:
([useragentpart=&quot;-webkit-media-text-track-display&quot;]):
(u):
(i):
(.hidden):

Also add missing .hidden to this legacy stylesheet.

([pseudo=&quot;-webkit-media-text-track-display&quot;] b): Deleted.
([pseudo=&quot;-webkit-media-text-track-display&quot;] u): Deleted.
([pseudo=&quot;-webkit-media-text-track-display&quot;] i): Deleted.

Canonical link: <a href="https://commits.webkit.org/275918@main">https://commits.webkit.org/275918@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ce034218ed69c1d7499e25625536032595b29d3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43288 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22320 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/45700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/45924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/39416 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45593 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26068 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19738 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/45924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43861 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/19355 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/45700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/45924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/16942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/45700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-skia~~](https://ews-build.webkit.org/#/builders/52/builds/1346 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/39484 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/45700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/47469 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/18204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/14992 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/47469 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/19741 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/45700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/47469 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9626 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/19920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/19372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
That is some fancy looking CSS.

annevk commented on Source/WebCore/css/mediaControls.css:
```diff
@@ -273,16 +273,11 @@ video::cue(:future) {
+    b { font-weight: bold; }
+    u { text-decoration: underline; }
+    i { font-style: italic; }
+    .hidden { display: none; }
```
Note that `.hidden` is new here. I guess that's fine, but maybe we should note it in the commit message.

anttijk committed:
REGRESSION(272743@main): WebVTT default styling is broken
https://bugs.webkit.org/show_bug.cgi?id=270783
rdar://124380882

Reviewed by Anne van Kesteren.

272743@main changed the attribute name for pseudo elements.
The WebVTT styling relied on those names.

* LayoutTests/media/track/captions-webvtt/inline.vtt: Added.
* LayoutTests/media/track/webvtt-inline-expected.txt: Added.
* LayoutTests/media/track/webvtt-inline.html: Added.

Add a test. We had no coverage.

* Source/WebCore/Modules/modern-media-controls/controls/text-tracks.css:
([useragentpart="-webkit-media-text-track-display"]):
(u):
(i):
(.hidden):
([pseudo="-webkit-media-text-track-display"] b): Deleted.
([pseudo="-webkit-media-text-track-display"] u): Deleted.
([pseudo="-webkit-media-text-track-display"] i): Deleted.
([pseudo="-webkit-media-text-track-display"] .hidden): Deleted.
* Source/WebCore/css/mediaControls.css:
([useragentpart="-webkit-media-text-track-display"]):
(u):
(i):
(.hidden):

Also add missing .hidden to this legacy stylesheet.

([pseudo="-webkit-media-text-track-display"] b): Deleted.
([pseudo="-webkit-media-text-track-display"] u): Deleted.
([pseudo="-webkit-media-text-track-display"] i): Deleted.

Canonical link: https://commits.webkit.org/275918@main
